apps/pipelines/nodes/nodes.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+from jinja2 import meta
+from jinja2.sandbox import SandboxedEnvironment
+from langchain_core.messages import BaseMessage
+from langchain_core.prompts import PromptTemplate
+from langchain_core.runnables import (
+    Runnable,
+    RunnableConfig,
+    RunnableLambda,
+)
+from langchain_core.runnables.utils import Input
+
+from apps.pipelines.exceptions import PipelineNodeBuildError
+from apps.pipelines.graph import Node
+from apps.pipelines.nodes.base import PipelineNode
+from apps.pipelines.nodes.types import LlmModel, LlmProviderId, LlmTemperature, PipelineJinjaTemplate
+from apps.pipelines.tasks import send_email_from_pipeline
+
+
+class RenderTemplate(PipelineNode):
+    template_string: PipelineJinjaTemplate
+
+    def get_runnable(self, node: Node) -> Runnable:
+        def fn(input: Input):
+            env = SandboxedEnvironment()
+            try:
+                if isinstance(input, BaseMessage):
+                    content = json.loads(input.content)
+                elif isinstance(input, dict):
+                    content = input
+                else:
+                    content = json.loads(input)
+            except json.JSONDecodeError:
+                # As a last resort, just set the all the variables in the template to the input
+                content = {var: input for var in meta.find_undeclared_variables(env.parse(self.template_string))}
+            template = SandboxedEnvironment().from_string(self.template_string)
+            return template.render(content)
+
+        return RunnableLambda(fn, name=self.__class__.__name__)
+
+
+class CreateReport(PipelineNode):
+    prompt: str = (
+        "Make a summary of the following text: {input}. "
+        "Output it as JSON with a single key called 'summary' with the summary."
+    )
+
+    def get_runnable(self, node: Node) -> Runnable:
+        return PromptTemplate.from_template(template=self.prompt) | LLMResponse.build(node)
```

Comment:
Since we're passing `node` to `LLMResponse.build` doesn't that imply that `node` should also contain the params for the `LLMResponse` node?

Just wondering if the `LLMResponse` attributes shouldn't be copied over or if `CreateReport` shouldn't extend `LLMResponse`.

Reply:
You are totally right. Before seeing this, I'd actually just removed the call to LLMResponse and added that in separately in the pipeline, but making it a subclass works well too - so I added it back in